Added client check to add the ability to run abapGit in production/QA environments

Srinu3366 opened the pull request:
Check if repository objects are modifiable in client. If not make write_protected setting active for all repos and make this setting disabled in repo setting page. This will not change this setting in ABAPGit database table.

Issue #2611

larshp commented:
cool, suggest adding the new method in class https://github.com/larshp/abapGit/blob/master/src/zcl_abapgit_environment.clas.abap instead of introducing a new class

Srinu3366 commented:
Done the changes as suggested. Removed the new class and added code to class `zcl_abapgit_environment`.

Now the changes include 

1. New method `is_repo_object_changes_allowed` in class `zcl_abapgit_environment`
2. Irrespective of saved settings enable write_protected in `zcl_abapgit_persistence_repo` if repository objects changes not allowed in the client. 
3. Disable the option to change write_protected in repo settings in class `zcl_abapgit_gui_page_repo_sett` if repository objects changes not allowed in the client. 
4. Remove the client check in ABAPGit main program include `ZABAPGIT_FORMS`
